src/kernel/mbrowse.ml

```diff
@@ -261,3 +277,39 @@ let print_node () node =
 
 let print () t =
   List.print (fun () (_,node) -> print_node () node) () t
+
+
+let check_node pos node =
+  let loc = node_merlin_loc node in 
+  if Location_aux.compare_pos pos loc = 0 then true else false 
+
+let get_children pos root = 
+  let children = 
+    match root with
+    | (_,Structure str) -> 
+        of_structure_items (List.filter ~f:(fun (_,x) ->
+            check_node pos (Structure_item (x, str.str_final_env))
+        ) (List.map ~f:(fun x -> (str.str_final_env, x)) str.str_items))
+    | (_,Signature str) -> 
+      of_signature_items (List.filter ~f:(fun (_,x) ->
+        check_node pos (Signature_item (x, str.sig_final_env))
+      ) (List.map ~f:(fun x -> (str.sig_final_env, x)) str.sig_items))
+    | _ -> raise (Invalid_argument "Not a valid root node")
```

Comment:
It's a strongly good practice to avoid raising. The reason: If this case is hit, then the user will see a merlin answer along the lines of "exception: Invalid_argument "Not a valid root node". That's not very helpful. (side-note: Luckily merlin catches exceptions at the end, so at least the server doesn't crash, and wraps the exception into an answer instead).

Instead, try to use the OCaml type system. Why don't you directly pass a structure or a signature to `get_children`? If you do so, that case can never be hit, and instead, we're protected by the type system to pass something to the function that it doesn't know to handle.